src/w3d/renderer/rendobj.cpp

```diff
@@ -107,6 +107,16 @@ RenderObjClass &RenderObjClass::RenderObjClass::operator=(const RenderObjClass &
     return *this;
 }
 
+RenderObjClass::~RenderObjClass()
+{
+    if (m_unknown) {
+        delete m_unknown;
+    }
+
+    // Thyme specific. Cleanup always to avoid dangling pointers.
+    Remove_Internal();
```

Comment:
do RenderObjClass::Remove instead

Reply:
I advise against this. Calling virtual function in destructor can lead to unexpected behaviour. It will make code less intuitive to understand, as you would need to exactly understand how destruction in virtual context works to anticipate the exact flow of execution that results from it. It is much safer to call non virtual function instead.

See:
https://www.artima.com/articles/never-call-virtual-functions-during-construction-or-destruction

Reply:
Ah. Meaning `RenderObjClass::Remove` will make it a CALL Fixed Address, yes?

Reply:
yes

Reply:
Ok we can do that. I will get back to it when the compile errors are fixed and I can test this change properly.

Reply:
Fixed.